cmd/state-offline-installer/install.go

```diff
@@ -60,17 +63,46 @@ func NewRunner(prime primeable) *runner {
 }
 
 type Params struct {
-	path string
+	path   string
+	config *InstallerConfig
 }
 
-func newParams() *Params {
-	return &Params{path: "/tmp"}
+func newParams(config *InstallerConfig) *Params {
+	return &Params{
+		path:   "/tmp",
+		config: config,
+	}
+}
+
+func (r *runner) Event(eventType string, installerDimensions *dimensions.Values) {
+	go r.analytics.Event(ac.CatRuntimeUsage, eventType, installerDimensions)
+}
+
+func (r *runner) EventWithLabel(eventType string, msg string, installerDimensions *dimensions.Values) {
+	go r.analytics.EventWithLabel(ac.CatRuntimeUsage, eventType, msg, installerDimensions)
+}
+
+func (r *runner) handleFailure(err error, msg string, installerDimensions *dimensions.Values) error {
+	r.EventWithLabel("failure", msg, installerDimensions)
+	return errs.Wrap(err, msg)
 }
 
 func (r *runner) Run(params *Params) error {
+	installerDimensions := &dimensions.Values{
+		ProjectID: params.config.ProjectID,
+		CommitID:  params.config.CommitID,
+		Trigger:   p.StrP(target.TriggerCliOfflineInstaller.String()),
+	}
+	r.analytics.Event(ac.CatRuntimeUsage, "start", installerDimensions)
```

Comment:
Right now I believe that `CatRuntimeUsage` is only used with heartbeats. I don't think that should be an issue but it's something to be mindful of. Maybe check with DW if that will affect anything?

Reply:
It's what @Naatan recommended we use for compatibility.  DW are on board with this too

Reply:
Offline installer now has its own category